src/server/apps/worldserver/worldserver.conf.dist

```diff
@@ -4153,3 +4153,14 @@ Metric.OverallStatusInterval = 1
 
 #
 ###################################################################################################
+#
+#    include ...
+#       Description: Include another configuration file. The paramaters from
+#                    that file will overwrite any conflicting parameters in this
+#                    file.
+#
+#include /etc/azerothcore/worldserver.conf
+#
+## To include a directory (and all files under it), make sure there's a slash at the end of the path
+#
+#include /etc/azerothcore/common.d/
```

Comment:
I would like this to follow the style used by conf files, and I would like it to be placed under AllowLoggingIPAddressesInDatabase option.

This is what I propse it to look like, but I am open to suggestions

```conf
#
#    ConfigurationInclude.file
#    ConfigurationInclude.dir
#        Description: The path to other configuration files to be loaded.
#                     Several files are seperated with a comma.
#                     A directory needs to end with a forward slash (/)
#        Example:     "C:/acore/conf/config1.conf"
#                     "C:/acore/conf/"
#        Default:     "" - (Disabled)
#

ConfigurationInclude.file = ""
ConfigurationInclude.dir  = ""
```

Reply:
I like the separate statement for specifying to include a directory. Makes the whole thing more clear. 

I prefer keeping the syntax for specifying an include different from the syntax for declaring a config parameter value. 

I'm thinking that it should be OK to include multiple files, such as 

```ini
include foo.conf
include bar.conf
```

The `include` statement is a property of syntax rather than a new configuration parameter and because of that, I think we should treat it like a property of the config files syntax. Having the `include` be declared the same way as a standard parameter but follow different rules may be confusing

```ini
# This is not OK, only the first parameter is read
DataDir = "/foo"
DataDir = "/bar"
# This is perfectly fine, both files get included
ConfigurationInclude.file = "foo.conf"
ConfigurationInclude.file = "bar.conf"
```

---

As far as placement in the config file for the documentation, I'm thinking after this section near the top of the file (where the syntax is described) makes the most sense

https://github.com/azerothcore/azerothcore-wotlk/blob/a4444a8b9d9d36d87b0f2a35f1a14ee2d9bd12da/src/server/apps/worldserver/worldserver.conf.dist#L40-L53

Reply:
Okay I get your point of having it seperate. Then I want it at the top of the file. But I still want it to follow the config style at the base.

My suggestion would be ... "bar.conf;foo.conf"

Reply:
so just to be clear, something could look like this? It's an extreme example that we (hopefully) won't ever see, but I'm just trying to make sure it looks how we want it too

```ini

# Multiple includes can be specified with a semicolon between then, or 
# separate include statements
include "/path/to/conf1.conf;/path/to/conf2.conf"
include "/path/to/conf3.conf"

# if foo or baz parameter occurs in the above config files, they will 
# be overwritten
foo = bar
baz = blah

# if foo or baz is in the below config files ,
# they will overwrite the foo or baz that happens above

# Directories can be included with the includeDir statement, and they 
# support semicolon as a separator as well
includeDir "/path/to/confDir"
include "/path/to/conf4.conf"
```

Reply:
Yep

Reply:
Looks good.

What happens if you have the same config option in 2 different files you include?

f.ex if I were to have WorldDatabaseInfo in conf1, conf2, worldserver.conf

Reply:
It's wherever it occurs "last". A way to think about `include` is that it functions as if you put all the parameters from the file you're referencing into the "root" file (worldserver.conf in this case). That said, duplication rules _in the same file_ still apply.


---

The final configuration value (and the one that is used) is `conf2`

<details>

<summary>worldserver.conf</summary>

```ini
include path/to/conf1.conf

WorldDatabaseInfo = "root"

include path/to/conf2.conf
```

</details>

<details>

<summary>path/to/conf1.conf</summary>

```ini
WorldDatabaseInfo = "conf1"
```

</details>

<details>

<summary>path/to/conf2.conf</summary>

```ini
WorldDatabaseInfo = "conf2"
```

</details>


This config "expands" out into something like this:

<details>

<summary>Config Expansion</summary>

```ini
WorldDatabaseInfo = "conf1"

WorldDatabaseInfo = "root"

WorldDatabaseInfo = "conf2"
```

</details>

---

The final configuration value is `prod`

<details>

<summary>worldserver.conf</summary>

```ini
WorldDatabaseInfo = "root"

include path/to/local.conf

include path/to/prod.conf
```

</details>

<details>

<summary>path/to/local.conf</summary>

```ini
WorldDatabaseInfo = "local"
```

</details>

<details>

<summary>path/to/prod.conf</summary>

```ini
WorldDatabaseInfo = "prod"
```

</details>

the `prod.conf` takes presedence, since it's where the last value comes from

<details>

<summary>Config Expansion</summary>

```ini
WorldDatabaseInfo = "root"

WorldDatabaseInfo = "local"

WorldDatabaseInfo = "prod"
```

</details>

---

The final configuration value is `conf2`

<details>

<summary>worldserver.conf</summary>

```ini
# worldserver.conf
WorldDatabaseInfo = "root"

include path/to/conf1.conf

include path/to/conf2.conf
```

</details>

<details>

<summary>path/to/conf1.conf</summary>

```ini
# test/to/conf1.conf
WorldDatabaseInfo = "conf1"
```

</details>

<details>

<summary>path/to/conf2.conf</summary>

```ini
# test/to/conf2.conf
WorldDatabaseInfo = "conf2"
```

</details>

Reply:
Also, most of the examples above are probably larger than we'd realistically see "in the wild". The main reason I wrote this PR is to help reduce the complexity/magic in the docker image build process:

<details>

<summary>Shell functions used in dashboard after compilation</summary>

https://github.com/azerothcore/azerothcore-wotlk/blob/026fe7c9627c01d10c674c6de5936ad2c1d0c94f/apps/compiler/includes/functions.sh#L189-L220

https://github.com/azerothcore/azerothcore-wotlk/blob/026fe7c9627c01d10c674c6de5936ad2c1d0c94f/apps/compiler/includes/functions.sh#L152-L169

</details>

This code controls how the files at `env/docker/etc/*.dockerdist` work. 

What happens is that after the docker build compiles, it will take all of the `dockerdist` files and do a find-and-replace so the proper values get set up in the `worldserver.conf`, `authserver.conf` that get created to start playing on the docker infra immediately. 

This works, but I think it's harder to maintain, there's more that can go wrong, and I think it's too much "extra" being done by the build process rather than doing this in code. I have a goal to greatly simplify the docker/docker-compose build process (see https://github.com/azerothcore/azerothcore-wotlk/pull/14990), and I think the changes in the this PR are needed as a part of that. 

With this PR, that ugly (I can call it ugly, I wrote it :p) function can be be reduced to this:

```bash
echo "include \"$(basename "dockerdist")\"" >> "$confDir/$filename"
```

Since this change is the underlying goal of the PR, I've included this change to help make the use of the feature clear and so I don't have to make another PR as soon as this one is merged, if it's merged.